ta4j-core/src/main/java/org/ta4j/core/analysis/criteria/ProfitLossPercentageCriterion.java

```diff
@@ -36,10 +36,20 @@
  */
 public class ProfitLossPercentageCriterion extends AbstractAnalysisCriterion {
 
+    private ProfitLossCriterion profitLoss = new ProfitLossCriterion();
+
     @Override
     public Num calculate(BarSeries series, TradingRecord tradingRecord) {
-        return tradingRecord.getTrades().stream().filter(Trade::isClosed).map(trade -> calculate(series, trade))
-                .reduce(series.numOf(0), Num::plus);
+        Num zero = series.numOf(0);
+        Num profit = profitLoss.calculate(series, tradingRecord);
+        Num purchaseCost = tradingRecord.getTrades().stream()
+            .filter(Trade::isClosed)
+            .map(trade -> series.getBar(trade.getEntry().getIndex()).getClosePrice())
```

Comment:
Summing up the `closePrice` is not correct, the `closePrice` is only the close price from the bar and nothing else. If you want to do it that way, then you must sum up the total cost of the trade (by using pricePerAsset, amount, costModel of the actual `Order.class`).

Reply:
@nimo23 I've added the trade amount into the mix. I don't think the cost model should be taken into account here, because that wasn't included in the original implementation, nor do any of the others do it.

However, I am working on a PR to make all the analysis implementations use the `Trade::getProfit()` method, rather than them all calculating it themselves in various ways.